src/main/java/io/divolte/server/filesinks/gcs/entities/ComposeRequest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 GoDataDriven B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server.filesinks.gcs.entities;
+
+import com.google.common.collect.ImmutableList;
+
+public class ComposeRequest {
```

Comment:
Not really. The fields are used for JSON serialisation, which allows nulls. The subsequent request to Google will result in an error in that case, but code-wise it's safe. It's just a value object to implement the JSON payload required for the call. Marking as nullable.